test/result/operator/math_operator.result

```diff
@@ -0,0 +1,44 @@
+select cast('2020-02-01' as date) + NULL;
+cast(2020-02-01 as date) + null
+null
+select 1 + NULL;
+1 + null
+null
+select NULL + 1;
+null + 1
+null
+select NULL + cast('2020-02-01 00:00:00' as datetime);
+null + cast(2020-02-01 00:00:00 as datetime(26))
+null
+drop table if exists t1;
+CREATE TABLE t1(
+cint int,
+cfloat float,
+str1 VARCHAR(20),
+str2 CHAR(20)
+);
+INSERT INTO t1 values(7, 5.3, '1', '1');
+INSERT INTO t1 values(7, 5.3, '0', '0');
+INSERT INTO t1 values(7, 5.3, '12', '0');
+INSERT INTO t1 values(7, 5.3, '10', '10');
+INSERT INTO t1 values(7, 5.3, '8.5', '1.2');
+INSERT INTO t1 values(7, 5.3, '1.5', '6.5');
+INSERT INTO t1 values(7, 5.3, '12.5', '3');
+INSERT INTO t1 values(7, 5.3, 'a', '2');
+INSERT INTO t1 values(7, 5.3, 'a', 'b');
+INSERT INTO t1 values(7, 5.3, 'xc', 'df');
+INSERT INTO t1 values(7, 5.3, 'xc', '1');
+select str1+str2 from t1;
+str1 + str2
+2.0
+0.0
+12.0
+20.0
+9.7
```

Comment:
Are you sure this is the right behavior?

select '8.5' + '1.2', should this be 9.7 or '8.51.2'?

Reply:
Yes, in MySQL, string addition converts string parameters into numbers, and then performs addition operations. It is not string concat.